File: test/three/app/src/main/java/com/z3d/material/DynamicConstItem.java
package com.z3d.material;

public class DynamicConstItem  extends DynamicBaseConstItem {

    public Curve curve ;


    public void update(float t ) {
        this.currentValue = this.curve.getValue(t);
        this.target.setDynamic(this);

    }
    public void update() {
        if (this.target!=null) {
            this.target.setDynamic(this);
        }
    }

    public void settype(int value) {
        this._type = value;
        this.curve = new Curve();
        this.curve.type = value;
    }

}

File: metalios/iosgl/iosgl/bt/material/DynamicConstItem.m
//
//  DynamicConstItem.m
//  iosgl
//
//  Created by zhao on 9/3/2020.
//  Copyright © 2020 zhao. All rights reserved.
//

#import "DynamicConstItem.h"

@implementation DynamicConstItem

- (instancetype)init
{
    self = [super init];
    if (self) {
        
    }
    return self;
}
-(void)update:(float)t;
{
   self. currentValue=   [self.curve getValue:t];
   [self.target setDynamic:self];
}
 

-(void)setType:(int)value;
{
    self._type=value;
    self.curve =[[Curve alloc]init];
    self.curve.type = value;
}
 
@end
